src/main/java/billgates/usecases/insert_entry/InsertEntryUseCase.java

```diff
@@ -0,0 +1,32 @@
+package billgates.usecases.insert_entry;
+
+import billgates.database.QueryEntryData;
+import billgates.entities.Entry;
+import billgates.entities.User;
+import billgates.interface_adapters.DatabaseGateway;
+import billgates.usecases.bill_update.BillUpdateUseCase;
+
+import java.util.List;
+
+public class InsertEntryUseCase implements InsertEntryInputPort{
+
+    private final InsertEntryOutputPort presenter;
+    private final DatabaseGateway gateway;
+    private final BillUpdateUseCase update;
+
+    public InsertEntryUseCase(InsertEntryOutputPort presenter, DatabaseGateway gateway, BillUpdateUseCase update){
+        this.presenter = presenter;
+        this.gateway = gateway;
+        this.update = update;
+    }
+
+    @Override
+    public void insertEntry(QueryEntryData entry) {
+        gateway.insertEntry(entry.getId(), entry);
+        update.updateBill();
```

Comment:
Similar problem as mentioned above, but what I'm stating might not be the final resolution yet, we have to reach a decision with Scott